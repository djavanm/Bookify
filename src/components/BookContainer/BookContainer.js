import React from 'react';
import Book from '../../containers/Book/Book'
import { connect } from 'react-redux';

const BookContainer = ({ books, toggleFavorite}) => {
  const displayedBooks = books.map(book => {
    return <Book data={book} key={book.book_id} toggleFavorite={toggleFavorite} />
  })

  return (
    <section className="book-container">
      {displayedBooks}
    </section>
  )
}

export const mapStateToProps = state => ({
  books: state.currentBooks
})

export default connect(mapStateToProps)(BookContainer);
