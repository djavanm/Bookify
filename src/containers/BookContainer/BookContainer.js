import React, { Component } from 'react';
import Book from '../Book/Book'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { showStart, showNext, showPrevious } from '../../actions';

const getTenBooks = (currentBooks, searchFilter) => {
  return currentBooks.slice(searchFilter.start, searchFilter.end)
}

const getFilteredFavorites = (favorites, genre) => {
  return genre ? favorites.filter(fav => fav.primary_genre_name === genre)
  : favorites;
};

const BookContainer = ({all, books, favorites, toggleFavorite, searchFilter, showStart, showNext, showPrevious}) => {
  const data = all ? books : favorites;
  const displayedBooks = data.map(book => {
    return <Book data={book} key={book.book_id} toggleFavorite={toggleFavorite} />
  })
  const showPrevBool = searchFilter.start === 0;
  const showNextBool = (searchFilter.length - searchFilter.start) < 10
  return (
    <>
    <section className='show-container' >
      { all && <button className='show-btn' onClick={() => showPrevious(searchFilter)} disabled={showPrevBool}>Show Prev</button> }
      { all && <button className='show-btn' onClick={() => showNext(searchFilter)} disabled={showNextBool}>Show Next</button> }
    </section>
    <section className='book-container'>
      {displayedBooks}
    </section>
    </>
  )
}

export const mapStateToProps = state => ({
  books: getTenBooks(state.currentBooks, state.searchFilter),
  favorites: getFilteredFavorites(state.favorites, state.genreFilter),
  genreFilter: state.genreFilter,
  searchFilter: state.searchFilter
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ showStart, showNext, showPrevious }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(BookContainer);
