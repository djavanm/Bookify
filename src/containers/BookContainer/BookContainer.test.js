import React from 'react';
import { shallow } from 'enzyme';
import { BookContainer } from './BookContainer';

describe('BookContainer', () => {
  let wrapper
  const booksMock = []
  const favoritesMock = [
    {
      author_name: 'Neil Gaiman',
      book_name: 'Good Omens',
      artwork_url: 'https://images/good',
      book_id: 42
    }
  ]
  const toggleFavoriteMock = jest.fn();
  const searchFilterMock = jest.fn();
  const showStartMock = jest.fn();
  const showNextMock = jest.fn();
  const showPreviousMock = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(<BookContainer
      all={true}
      books={booksMock}
      favorites={favoritesMock}
      toggleFavorite={toggleFavoriteMock}
      searchFilter={searchFilterMock}
      showStart={showStartMock}
      showNext={showNextMock}
      showPrevious={showPreviousMock}
      />)
  })

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})