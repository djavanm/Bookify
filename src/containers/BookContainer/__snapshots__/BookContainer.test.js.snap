// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContainer should match the snapshot with the data passed through 1`] = `
<section
  className="containers"
>
  <section
    className="show-container"
  >
    <button
      className="show-btn"
      disabled={false}
      onClick={[Function]}
    >
      Show Prev
    </button>
    <button
      className="show-btn"
      disabled={false}
      onClick={[Function]}
    >
      Show Next
    </button>
  </section>
  <section
    className="book-container"
  />
</section>
`;
